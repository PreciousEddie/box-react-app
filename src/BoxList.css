.BoxList-form {
  margin: 15px;
}
