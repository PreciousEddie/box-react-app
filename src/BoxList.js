import React, { useState } from "react";
import { v4 as uuid } from "uuid"
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";
import "./BoxList.css"

const BoxList = () => {
    const INITIAL_STATE = [];
    const [boxes, setBoxes] = useState(INITIAL_STATE);
    const addBox = (newBox) => {
        setBoxes(boxes => [...boxes, { ...newBox, id: uuid() }]);
    };
    const removeBox = (id) => {
        setBoxes((boxes) => boxes.filter((box) => box.id !== id));
    };
    return (
        <>
            <div className="BoxList-form">
                <NewBoxForm addBox={ addBox } />
            </div>
            <div>
                { boxes.map(({ id, width, height, backgroundColor }) => (
                    <Box
                        id={ id }
                        key={ id }
                        width={ width }
                        height={ height }
                        backgroundColor={ backgroundColor }
                        removeBox={ removeBox }
                    />
                )) }
            </div>
        </>
    )
}

export default BoxList;