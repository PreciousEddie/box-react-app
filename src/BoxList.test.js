import { render, screen, fireEvent } from '@testing-library/react';
import BoxList from './BoxList';

it('renders without crashing', function () {
    render(<BoxList />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot();
});

test('adding a box by submitting the form', () => {
    render(<BoxList />);

    // Get the input fields
    const widthInput = screen.getByLabelText(/Width:/i);
    const heightInput = screen.getByLabelText(/Height:/i);
    const colorInput = screen.getByLabelText(/Color:/i);

    // Simulate user input
    fireEvent.change(widthInput, { target: { value: '200' } });
    fireEvent.change(heightInput, { target: { value: '150' } });
    fireEvent.change(colorInput, { target: { value: 'red' } });

    // Submit the form
    const createButton = screen.getByText(/Create Box!/i);
    fireEvent.click(createButton);

    // Verify if the box is added to the DOM
    const boxElement = screen.getByTestId('box-element');
    expect(boxElement).toBeInTheDocument();
});