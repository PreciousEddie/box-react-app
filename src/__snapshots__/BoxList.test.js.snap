// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList-form"
  >
    <form>
      <label
        for="width"
      >
        Width:
      </label>
      <input
        id="width"
        max="500"
        min="100"
        name="width"
        placeholder="Width"
        type="number"
        value=""
      />
      <label
        for="height"
      >
        Height:
      </label>
      <input
        id="height"
        max="500"
        min="100"
        name="height"
        placeholder="Height"
        type="number"
        value=""
      />
      <label
        for="backgroundColor"
      >
        Color:
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        placeholder="Choose a color"
        type="text"
        value=""
      />
      <button>
        Create Box!
      </button>
    </form>
  </div>
  <div />
</DocumentFragment>
`;
